docs/reconcile-in-custom-namespace.md

```diff
@@ -0,0 +1,53 @@
+# Reconcile Cluster-API Objects In Custom Namespace <!-- omit in toc -->
+
+Cluster-api-provider-aws controllers by default, reconcile cluster-api objects
+across all namespaces in the cluster. However, it is possible to restrict
+reconciliation to a single namespace and this document tells you how.
+
+## Contents <!-- omit in toc -->
+
+- [Use cases](#use-cases)
+- [Configuring `cluster-api-provider-aws` controllers](#configuring-cluster-api-provider-aws-controllers)
+
+## Use cases
+
+- Grouping clusters into a namespace based on the AWS account will allow
+  managing clusters across multiple AWS accounts. This will require each
+  `cluster-api-provider-aws` controller to have credentials to their respective
+  AWS accounts. These credentials can be created as kubernetes secret and be
+  mounted in the pod at `/root/.aws`.
+- Grouping clusters into a namespace based on their environment, (test,
+  qualification, canary, production) will allow a phased rolling out
+  `cluster-api-provider-aws` releases.
+- Grouping clusters into a namespace based on the infrastructure provider will
+  allow running multiple cluster-api provider implementations side-by-side and
+  manage clusters across infrastructure providers.
+
+## Configuring `cluster-api-provider-aws` controllers
+
+- Create the namespace that `cluster-api-provider-aws` controller will watch for
+  cluster-api objects
+
+```(bash)
+cat <<EOF | kubectl apply -f -
+apiVersion: v1
+kind: Namespace
+metadata:
+  labels:
+  name: my-pet-clusters #edit if necessary
```

Comment:
```suggestion
    name: my-pet-clusters #edit if necessary
```

Reply:
I think the indentation is correct, but `labels:` should probably be omitted.

Reply:
oh ofc! good catch

Reply:
Yes, the `labels` is stale from copy-pasta.
Fixed now.